src/backwards_compatibility.jl

```diff
@@ -35,3 +35,14 @@ end
         return isa(t, DataType) && t.mutable
     end
 end
+
+@static if :ninitialized in fieldnames(Tuple{})
+    # https://github.com/JuliaIO/JLD2.jl/issues/327
+    function ninitialized(@nospecialize(T::Type))::Int
+        T.ninitialized
+    end
+else
+    function ninitialized(@nospecialize(T::Type))::Int
```

Comment:
Not sure if there is a benefit of `@nospecialize` here, again I just followed the tradition `function ismutabletype(@nospecialize(t::Type))`